### PR TITLE
Add optional replacement to `ForbiddenDirectives` 

### LINKED ISSUE #1900: Add replacement options to `ForbiddenDirectives` rule
Follows https://github.com/symfony/symfony-docs/issues/20371#issuecomment-2525050843

cc @TimoBakx

### PR DESCRIPTION
**Pull request by TimoBakx**
Closes #1900

### CODE REVIEW

**Comment by OskarStark** [comment]
After this PR is merged, I would please a new version, then we need to wait ~10 minutes and then a PR against `symfony/symofny-docs:@6.4` would be the next step, raising the version in the GithubActions workflow file + adjusting the `.doctor-rst.yaml` file accordingly.

File: src/Rule/ForbiddenDirectives.php [hunk 1]
```diff
@@ -29,15 +30,24 @@ class ForbiddenDirectives extends AbstractRule implements Configurable, LineCont
     use DirectiveTrait;
 
     /**
-     * @var array<string>
+     * @var array<array[directive<string>, replacement<?string>]>
```
[the remark this answers is not in the document]
Comment (wouterj) [reply src/Rule/ForbiddenDirectives.php:33]:
```suggestion
     * @var array<array{directive: string, replacement: ?string}>
```

[the remark this answers is not in the document]
Comment (OskarStark) [reply src/Rule/ForbiddenDirectives.php:33]: I don't think so @wouterj

Can you maybe add a test for the configuration?

**Comment by TimoBakx** [comment]
Updated the code so it takes an array of replacements. Refactored the tests a bit because there are a lot more variations now. Also added tests for configuration failures.

**Comment by OskarStark** [comment]
Thanks @TimoBakx 

You may want to send a PR to `symfony-docs` like mentioned before?

**Comment by TimoBakx** [comment]
Yes, I will do that.

**Pull request merged**